### PR TITLE
An attempt to add limit/offset to history contents

### PR DESCRIPTION
**Pull request by carlfeberhard**
...by implementing a history_contents manager which can filter, order, and limit/offset a combined (composite) list of both HDAs and HDCAs. 

The main interface is the `.contents` method and this is meant to surround an (overly) complex strategy that's (AFAIK) necessary due to the relation between histories, HDAs, and HDCAs. See [below](https://github.com/galaxyproject/galaxy/compare/dev...carlfeberhard:managers.history-contents?expand=1#diff-3e81b262b07e5e9ecc994f6eed8ba400R90) for more info. I'm open to suggestions on simplifying _how_ the behavior works behind `.contents`, but at least now it's cordoned off.

The history_contents manager will probably be modified to a fuller composite pattern over the two history subclasses. Then it might be able to inform the design of the `ContainerManagerMixin` a bit more. From there to the history_contents API.

WIP and all that.

### CODE REVIEW

**Comment by carlfeberhard** [comment]
```
./test/unit/managers/test_HistoryManager.py:7:1: F401 'random' imported but unused
./test/unit/managers/test_HistoryManager.py:20:1: F401 'CreatesCollectionsMixin' imported but unused
./test/unit/managers/test_HistoryManager.py:26:1: F401 'collections' imported but unused
```

And yet they are...

**Comment by carlfeberhard** [comment]
Oh! test__History_Manager...Jeez - I really am pretty dense sometimes.

File: test/unit/managers/test_HistoryContentsManager.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+import os
+import imp
+import unittest
+import random
+
+test_utils = imp.load_source( 'test_utils',
+    os.path.join( os.path.dirname( __file__), '../unittest_utils/utility.py' ) )
+
+from galaxy import eggs
+eggs.require( 'SQLAlchemy >= 0.4' )
```
Comment (jmchilton) [inline test/unit/managers/test_HistoryContentsManager.py]: This can/should be dropped now.

**Comment by martenson** [comment]
thanks @carlfeberhard !

**Pull request merged**